### PR TITLE
ISPN-15408 Remove use of ProtoStream MessageMarshaller

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-15408

This PR removes all ProtoStream `MessageMarshaller` implementations as it will be removed in ProtoStream 4.5.x.

@fax4ever Can you review this PR as it mainly affects query.

### CODE REVIEW

**Review by fax4ever — APPROVED** [review]
Great job.
I'm very happy we've freed 2.4K LOC removing the `MessageMarshaller`s.
I have just few comments.
+1 to merge as soon as we fix the CI on GraalVM HotRod client test

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/marshall/EmbeddedRemoteInteropQueryTest.java [hunk 1]
```diff
@@ -165,6 +165,7 @@ public void testRemoteQueryForEmbeddedEntry() {
       assertAccount(list.get(0), AccountPB.class);
    }
 
+   // TODO Failing
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/marshall/EmbeddedRemoteInteropQueryTest.java:168]: Nope. I'll remove

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/marshall/EmbeddedRemoteInteropQueryTest.java [hunk 2]
```diff
@@ -496,27 +498,22 @@ private void assertAccount(Account account, Class<?> cls) {
...
+               TransactionHS.class,
+               UserHS.class,
+               User.Gender.class
+         },
+         schemaFileName = "test.client.EmbeddedRemoteInteropQueryTest.proto",
+         schemaFilePath = "proto/generated",
+         schemaPackageName = "sample_bank_account",
+         service = false
+   )
+   interface ServerSCI extends SerializationContextInitializer {
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/marshall/EmbeddedRemoteInteropQueryTest.java:517]: `DslSCI` already exists in the `query` module for the `*HS` classes, however it has a different `schemaPackageName` and we require these types to have "sample_bank_account", so we must create a new one specific for this test.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**